cli/src/main.rs

```diff
@@ -428,6 +433,53 @@ fn run() -> Result<()> {
             }
         }
 
+        Commands::Build(build_options) => {
+            if build_options.wasm {
+                let grammar_path =
+                    current_dir.join(build_options.path.as_deref().unwrap_or_default());
+                let (output_dir, output_path) = if let Some(ref path) = build_options.output {
+                    (current_dir.clone(), Some(current_dir.join(path)))
+                } else {
+                    (loader.parser_lib_path.clone(), None)
+                };
+                wasm::compile_language_to_wasm(
+                    &loader,
+                    &grammar_path,
+                    &output_dir,
+                    output_path,
+                    build_options.docker,
+                )?;
+            } else {
+                let grammar_path =
+                    current_dir.join(build_options.path.as_deref().unwrap_or_default());
+                let output_path = if let Some(ref path) = build_options.output {
+                    let path = Path::new(path);
+                    if path.is_absolute() {
+                        path.to_path_buf()
+                    } else {
+                        current_dir.join(path)
+                    }
+                } else {
+                    let file_name = grammar_path
+                        .file_stem()
+                        .unwrap()
+                        .to_str()
+                        .unwrap()
+                        .strip_prefix("tree-sitter-")
+                        .unwrap_or("parser");
+                    current_dir
+                        .join(file_name)
+                        .with_extension(env::consts::DLL_EXTENSION)
+                };
+                let config = Config::load(None)?;
+                let loader_config = config.get()?;
+                loader.find_all_languages(&loader_config).unwrap();
+                loader
+                    .compile_parser_at_path(&grammar_path, output_path)
```

Comment:
This ultimately calls `compile_parser_to_dylib` which defines `TREE_SITTER_INTERNAL_BUILD` and `TREE_SITTER_REUSE_ALLOCATOR` which should only be defined when compiled for internal use.

Reply:
they're available as optional flags (a user might want to compile a parser while reusing the allocator), but aren't set by default when invoked w/ the cli's build command